libcaf_core/tests/legacy/detail/parser/read_number_or_timespan.cpp

```diff
@@ -29,7 +29,10 @@ struct number_or_timespan_parser_consumer {
 
 struct res_t {
   std::variant<pec, double, int64_t, timespan> val;
-  template <class T>
+  res_t(const res_t&) = default;
+  res_t(res_t&&) = default;
+  template <class T,
+            class E = std::enable_if_t<!std::is_same_v<std::decay_t<T>, res_t>>>
```

Comment:
Same as the other `res_t`: primitive types don't need all of this move/forward complexity.